vdev/src/testing/integration.rs

```diff
@@ -64,7 +64,23 @@ impl IntegrationTest {
         for (key, value) in config_env(&self.env_config) {
             env_vars.insert(key, Some(value));
         }
-        let mut args = self.config.args.clone();
+
+        let mut args = self.config.args.clone().unwrap_or_default();
+
+        args.push("--features".to_string());
+        args.push(self.config.features.join(","));
+
+        // If the test field is not present then use the --lib flag
+        match self.config.test {
+            Some(ref test_arg) => {
+                args.push("--test".to_string());
+                args.push(test_arg.to_string());
+            }
+            None => args.push("--lib".to_string()),
+        }
+
+        // Ensure the test_filter args are passed as well
+        args.push(self.config.test_filter.clone().unwrap_or("".to_string()));
```

Comment:
If `test_filter` is `None`, we want to skip pushing the arg, not push an empty string.